src/pulse3D/excel_writer.py

```diff
@@ -362,21 +364,31 @@ def write_xlsx(
         min_value = min(interpolated_force)
         interpolated_force -= min_value
         interpolated_force *= MICRO_TO_BASE_CONVERSION
-        #find the biggest activation twitch force over all
-
+        
+        # find the biggest activation twitch force over all
         max_force_of_well = max(interpolated_force)
         max_force_of_recording = max(max_force_of_recording, max_force_of_well)
+
         try:
             # compute peaks / valleys on interpolated well data
             log.info(f"Finding peaks and valleys for well {well_name}")
-            peaks_and_valleys = peak_detector(
-                interpolated_well_data,
-                prominence_factors=prominence_factors,
-                width_factors=width_factors,
-                start_time=start_time,
-                end_time=end_time,
-            )
 
+            if peaks_valleys is None:
+                log.info(f"No user defined peaks and valleys were found, so calculating with peak_detector")
+                peaks_and_valleys = peak_detector(
+                    interpolated_well_data,
+                    prominence_factors=prominence_factors,
+                    width_factors=width_factors,
+                    start_time=start_time,
+                    end_time=end_time,
+                )
+            else:
+                # convert peak and valley lists into a format compatible with find_twitch_indices
+                peaks, valleys = [np.array(i) for i in peaks_valleys.get(well_name)]
```

Comment:
If `peak_valleys` is a dict, then is using `peaks_valleys.get(well_name)` instead of `peaks_valleys[well_name]` necessary? To me it looks like if the key isn't present an error is going to get raised here either way, either `TypeError: 'NoneType' object is not iterable` if you use `get` or a `KeyError` if you don't. Also what is `i` here?